packages/frontend/src/features/metricsCatalog/components/MetricsTable.tsx

```diff
@@ -63,6 +71,25 @@ export const MetricsTable = () => {
         [data],
     );
 
+    // Check if we are mutating any of the icons or categories related mutations
+    const isMutating = useIsMutating({
+        predicate: (mutation) => {
+            const mutationKeys = [
+                'create-tag',
+                'update-tag',
+                'delete-tag',
+                'add-category',
+                'remove-category',
+                'update-catalog-item-icon',
+            ];
+            return Boolean(
+                mutation.options.mutationKey?.some((key) =>
+                    mutationKeys.includes(key as string),
+                ),
+            );
+        },
+    });
```

Comment:
can be moved to a separate hook? 

wonder how cache keys can be maintained better. constants?

Reply:
I think having the keys as constants would be a very nice addition here. Don't think we would need to extract this to a separate hook for now, if we end up reusing it then I would do that